webapi/src/main/scala/org/knora/webapi/slice/admin/repo/service/AbstractEntityRepo.scala

```diff
@@ -148,3 +148,15 @@ abstract class AbstractEntityRepo[E <: EntityWithId[Id], Id <: StringValue](
     Update(query.getQueryString)
   }
 }
+
+abstract class CachingEntityRepo[E <: EntityWithId[Id], Id <: StringValue](
+  triplestore: TriplestoreService,
+  mapper: RdfEntityMapper[E],
+  private val cache: EntityCache[Id, E],
+) extends AbstractEntityRepo[E, Id](triplestore, mapper) {
+
+  override def findById(id: Id): Task[Option[E]] =
+    cache.get(id).fold(super.findById(id).map(_.map(cache.put)))(ZIO.some(_))
+
+  override def save(entity: E): Task[E] = super.save(entity).map(cache.put)
+}
```

Comment:
This might be better put in its own file, which is according to the style guide. https://docs.scala-lang.org/style/files.html

Reply:
Agreed: => https://github.com/dasch-swiss/dsp-api/pull/3200/commits/5fe76ed3f9ac1d8b1e31b9d18ac205afe0148de0